src/pkt_proc.c

```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
     const void *transport_start;
     struct flow_key key;
     
-	memset(&key, 0x00, sizeof(struct flow_key));
+    memset(&key, 0x00, sizeof(struct flow_key));
 
     flocap_stats_incr_num_packets();
     fprintf(info, "\n");
     joy_log_info("++++++++++ Packet %lu ++++++++++", flocap_stats_get_num_packets());
     //  packet_count++;
   
     // ethernet = (struct ethernet_hdr*)(packet);
-  
-    /* define/compute ip header offset */
-    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
```

Comment:
A comment here explaining the purpose of this code block, i.e. "Support of both normal ethernet and 802.1q. Distinguish between the two accepted types."